### PR TITLE
Fix typos and add spelling workflow

### PR DESCRIPTION
**Pull request by szepeviktor**
From https://github.com/szepeviktor/typos-on-you/actions/runs/5435919690/jobs/9885404517

### CODE REVIEW

File: .github/workflows/pr-ci.yml [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow
```
Comment (szepeviktor) [inline .github/workflows/pr-ci.yml:1]: This helps modern IDE-s to validate workflows.

File: .github/workflows/pr-ci.yml [hunk 2]
```diff
@@ -6,6 +8,9 @@ on:
   pull_request:
     branches:
       - master
+concurrency:
+  group: '${{ github.workflow }}-${{ github.ref }}'
+  cancel-in-progress: true
```
Comment (szepeviktor) [inline .github/workflows/pr-ci.yml:13]: Stop previous runs in case of a new commit.

File: tests/t3_fs_api.py [hunk 3]
```diff
@@ -218,7 +218,7 @@ async def test_getxattr(ctx):
     await ctx.server.release(fh)
 
     with assert_raises(FUSEError):
-        await ctx.server.getxattr(inode.st_ino, b'nonexistant-attr', some_ctx)
+        await ctx.server.getxattr(inode.st_ino, b'nonexistent-attr', some_ctx)
```
Comment (szepeviktor) [inline tests/t3_fs_api.py:221]: This may need your attention.

File: .github/workflows/spelling.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    branches:
+      - master
+permissions:
+  contents: read
+concurrency:
+  group: '${{ github.workflow }}-${{ github.ref }}'
+  cancel-in-progress: true
+jobs:
+  typos_check:
+    name: '文A Typos check'
```
Comment (Nikratio) [inline .github/workflows/spelling.yml:18]: Some odd character here (renders as a square for me)?

Comment (szepeviktor) [reply .github/workflows/spelling.yml:18]: It tries to resemble https://commons.wikimedia.org/wiki/File:Translation_icon_cropped.svg

Removing ...

Comment (szepeviktor) [reply .github/workflows/spelling.yml]: BTW Are you running Linux that https://en.wiktionary.org/wiki/%E6%96%87 is not included in your font?

**Pull request merged**